### PR TITLE
improved API

### PR DESCRIPTION
**Pull request by pelger**

### CODE REVIEW

File: adapters/balance.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ var uuid = require('uuid')
  * Load balance transport adapter.
  * distributes requests round robin to each supplied transport
  */
-module.exports = function (mu, transports) {
+module.exports = function (transports) {
   var muid = uuid()
```
Comment (lucamaraschi) [inline adapters/balance.js]: Didn't we settle in using [hyperid](https://github.com/mcollina/hyperid)?

File: adapters/balance.js [hunk 2]
```diff
@@ -52,13 +60,8 @@ module.exports = function (mu, transports) {
     muid: muid,
     tf: tf,
     type: 'transport',
-    setId: setId
-    // driver: driver,
-    // tearDown: tearDown
+    setId: setId,
```
Comment (lucamaraschi) [inline adapters/balance.js]: @pelger can you please elaborate on the overall signature of this object?I am a little bit confused by the object contract by the using of explicit *setter* (or look alike!) functions.

File: adapters/balance.js [hunk 3]
```diff
@@ -38,6 +38,14 @@ module.exports = function (mu, transports) {
 
 
 
+  function setMu (muInstance) {
+    transports.forEach(function (transport) {
+      transport.setMu(muInstance)
+    })
+  }
```
Comment (mcollina) [inline adapters/balance.js]: Why this? I think we can avoid setMu for just a transport.mu = mu. The setter can be overloaded if needed.

The same holds true for the id.

Comment (mcollina) [inline adapters/balance.js]: Also, I don't think outbound transports needs access to the mu instance.

File: adapters/balance.js (hunk 1 above)
Comment (pelger) [inline adapters/balance.js]: Just trying to get the structure agreed - we can switch this in at any point

File: adapters/balance.js (hunk 3 above)
Comment (pelger) [inline adapters/balance.js]: The mu instance is in the closure scope which is why the setter is required. Outbound does require access to the mu instance!

File: adapters/balance.js (hunk 2 above)
Comment (pelger) [inline adapters/balance.js]: Should have removed the commented out line - will do that! Adapters present the same interface as transport.js. i.e. an adapter looks exactly like a transport to the router - make sense ?

**Comment by pelger** [comment]
Guys would like to merge this - can you confirm pls @mcollina @lucamaraschi

**Comment by lucamaraschi** [comment]
👍 LGTM
I guess @pelger just wants to get a basic structure to start making pull requests ;-)

File: adapters/balance.js (hunk 3 above)
Comment (mcollina) [inline adapters/balance.js]: why? Just saying, outbound should be something that calls `mu`.

Anyway, we can always change this later on.

Comment (pelger) [inline adapters/balance.js]: I think it is necessary for the transport to hold a reference to the mu scope, however this is an internal detail which may change

**Pull request merged**